src/ucp/api/ucp.h

```diff
@@ -1906,6 +1906,32 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
 ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ *
+ * @brief Non-blocking @ref ucp_ep_h "endpoint" closure.
+ *
+ * This routine releases the @ref ucp_ep_h "endpoint". The endpoint closure
+ * process depends on the selected @a mode.
+ *
+ * @param [in]  ep      Handle to the endpoint to close.
+ * @param [in]  mode    One from @ref ucp_ep_close_mode value.
+ * @param [in]  param   Operation parameters, see @ref ucp_request_param_t
+ *
+ * @return UCS_OK           - The endpoint is closed successfully.
+ * @return UCS_PTR_IS_ERR(_ptr) - The closure failed and an error code indicates
+ *                                the transport level status. However, resources
```

Comment:
if cb is specified, will it be called in this case?

Reply:
this is API proposal only, not behavior

Reply:
Yes, but we already have some flags which are not relevant for this func (datatype).
I think we have to mention what parameters are relevant (or not relevant) for every func accepting new parameters struct.

Reply:
right, but final doc (parameters used, description parameter) depends from implementation and doc format (where params values are described). this is task for tech writers mostly, here I would like to identify API itself only

Reply:
@hoopoepg i think it's best to document